app/src/ui/diff/side-by-side-diff.tsx

```diff
@@ -370,8 +362,13 @@ export class SideBySideDiff extends React.Component<
   }
 
   private async initDiffSyntaxMode() {
-    const { file, repository } = this.props
-    const { contents: currentContents, diff: currentDiff } = this.state.document
+    const contents = this.props.fileContents
+
+    if (contents === null) {
+      return
+    }
```

Comment:
I believe there's a change in behavior here. Previously we'd always syntax highlight the first 256kb of file contents and then bail if the file was longer. With this it looks like we skip on highlighting entirely if the file cannot be expanded (meaning it's 256kb  - 1b or longer).

I don't think whether a file can be expanded should be coupled to syntax highlighting

Reply:
Ouch! That's completely true, I'll fix it 😳 
Nice catch!

Reply:
@niik the last set of commits should fix this. Basically this check here makes sense, the problem is I was only passing down the `fileContents` when the diff can be expanded. Changing that logic required changing a few other checks here and there.

I also realized this new approach broke syntax highlighting for huge files (> 10MB), so I fixed that too. The only caveat of this is that we'll load the 0.5MB of file contents for huge files before the user clicked on `Show Diff`, but I think that should be fine since it won't be a very common scenario and, again, it's only 0.5MB. Does that sound reasonable to you?